shiny/_main.py

```diff
@@ -626,3 +629,47 @@ class ReloadArgs(TypedDict):
     reload_includes: NotRequired[list[str]]
     reload_excludes: NotRequired[list[str]]
     reload_dirs: NotRequired[list[str]]
+
+
+# Check that the version of rsconnect supports Shiny Express; can be removed in the
+# future once this version of rsconnect is widely used. (Added 2024-03)
+def _verify_rsconnect_version() -> None:
+    PACKAGE_NAME = "rsconnect-python"
+    MIN_VERSION = "1.22.0"
+
+    from importlib.metadata import PackageNotFoundError, version
+
+    def _safe_int(x: str) -> int:
```

Comment:
I think we already have an indirect dependency on `packaging`, do we want to make it direct  and use their version parsing? 

```
    from packaging.version import parse
    from importlib.metadata import PackageNotFoundError, version
    import sys

    try:
        installed_version = parse(version(PACKAGE_NAME))
        required_version = parse(MIN_VERSION)
        if installed_version < required_version:
      ...
```

Reply:
Good call -- I see that htmltools uses `packaging` to check versions, so I've updated the code as you suggested.